encodings/zstd/src/array.rs

```diff
@@ -88,11 +94,46 @@ fn choose_max_dict_size(uncompressed_size: usize) -> usize {
     (uncompressed_size / 100).clamp(256, 100 * 1024)
 }
 
-fn collect_valid(parray: &PrimitiveArray) -> VortexResult<PrimitiveArray> {
+fn collect_valid_primitive(parray: &PrimitiveArray) -> VortexResult<PrimitiveArray> {
     let mask = parray.validity_mask()?;
     filter(&parray.to_array(), &mask)?.to_primitive()
 }
 
+fn collect_valid_vbv(vbv: &VarBinViewArray) -> VortexResult<(Buffer<u8>, Vec<usize>)> {
+    let mask = vbv.validity_mask()?;
+    let mut buffer = BufferMut::empty();
+    let mut value_byte_indices = Vec::new();
+    for i in 0..vbv.len() {
+        if mask.value(i) {
```

Comment:
I think you might want `match mask.boolean_buffer() { AllTrue ... AllFalse .. BooleanBuffer }` to handle fast paths. BooleanBuffer can then use an iterator to avoid a function call for each value.

Reply:
Done, but there's no way to make a fast path for `All`.